plugins/defaultplugins/aclplugin/vppcalls/acl_vppcalls_test.go

```diff
@@ -0,0 +1,490 @@
+					SourceNetwork:      "",
+					DestinationNetwork: "",
+				},
+			},
+		},
+	},
+	{
+		RuleName:  "denyICMP",
+		AclAction: acl.AclAction_DENY,
+		Match: &acl.AccessLists_Acl_Rule_Match{
+			IpRule: &acl.AccessLists_Acl_Rule_Match_IpRule{
+				Icmp: &acl.AccessLists_Acl_Rule_Match_IpRule_Icmp{
+					Icmpv6: false,
+					IcmpCodeRange: &acl.AccessLists_Acl_Rule_Match_IpRule_Icmp_Range{
+						First: 150,
+						Last:  250,
+					},
+					IcmpTypeRange: &acl.AccessLists_Acl_Rule_Match_IpRule_Icmp_Range{
+						First: 1150,
+						Last:  1250,
+					},
+				},
+			},
+		},
+	},
+	{
+		RuleName:  "denyICMPv6",
+		AclAction: acl.AclAction_DENY,
+		Match: &acl.AccessLists_Acl_Rule_Match{
+			IpRule: &acl.AccessLists_Acl_Rule_Match_IpRule{
+				Icmp: &acl.AccessLists_Acl_Rule_Match_IpRule_Icmp{
+					Icmpv6: true,
+					IcmpCodeRange: &acl.AccessLists_Acl_Rule_Match_IpRule_Icmp_Range{
+						First: 150,
+						Last:  250,
+					},
+					IcmpTypeRange: &acl.AccessLists_Acl_Rule_Match_IpRule_Icmp_Range{
+						First: 1150,
+						Last:  1250,
+					},
+				},
+			},
+		},
+	},
+	{
+		RuleName:  "permitTCP",
+		AclAction: acl.AclAction_PERMIT,
+		Match: &acl.AccessLists_Acl_Rule_Match{
+			IpRule: &acl.AccessLists_Acl_Rule_Match_IpRule{
+				Tcp: &acl.AccessLists_Acl_Rule_Match_IpRule_Tcp{
+					TcpFlagsMask:  20,
+					TcpFlagsValue: 10,
+					SourcePortRange: &acl.AccessLists_Acl_Rule_Match_IpRule_PortRange{
+						LowerPort: 150,
+						UpperPort: 250,
+					},
+					DestinationPortRange: &acl.AccessLists_Acl_Rule_Match_IpRule_PortRange{
+						LowerPort: 1150,
+						UpperPort: 1250,
+					},
+				},
+			},
+		},
+	},
+	{
+		RuleName:  "denyUDP",
+		AclAction: acl.AclAction_DENY,
+		Match: &acl.AccessLists_Acl_Rule_Match{
+			IpRule: &acl.AccessLists_Acl_Rule_Match_IpRule{
+				Udp: &acl.AccessLists_Acl_Rule_Match_IpRule_Udp{
+					SourcePortRange: &acl.AccessLists_Acl_Rule_Match_IpRule_PortRange{
+						LowerPort: 150,
+						UpperPort: 250,
+					},
+					DestinationPortRange: &acl.AccessLists_Acl_Rule_Match_IpRule_PortRange{
+						LowerPort: 1150,
+						UpperPort: 1250,
+					},
+				},
+			},
+		},
+	},
+}
+
+var acl_MACIPrules = []*acl.AccessLists_Acl_Rule{
+	{
+		RuleName:  "denyIPv4",
+		AclAction: acl.AclAction_DENY,
+		Match: &acl.AccessLists_Acl_Rule_Match{
+			MacipRule: &acl.AccessLists_Acl_Rule_Match_MacIpRule{
+				SourceAddress:        "192.168.0.1",
+				SourceAddressPrefix:  uint32(16),
+				SourceMacAddress:     "11:44:0A:B8:4A:35",
+				SourceMacAddressMask: "ff:ff:ff:ff:00:00",
+			},
+		},
+	},
+	{
+		RuleName:  "denyIPv6",
+		AclAction: acl.AclAction_DENY,
+		Match: &acl.AccessLists_Acl_Rule_Match{
+			MacipRule: &acl.AccessLists_Acl_Rule_Match_MacIpRule{
+				SourceAddress:        "dead::1",
+				SourceAddressPrefix:  uint32(64),
+				SourceMacAddress:     "11:44:0A:B8:4A:35",
+				SourceMacAddressMask: "ff:ff:ff:ff:00:00",
+			},
+		},
+	},
+}
+
+func TestAddIPAcl(t *testing.T) {
```

Comment:
a short comment which explains what the test is doing could be very helpful if there will be a need to modify it in the future. Please add to every test case